src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -161,6 +161,23 @@ program::program(engine& engine_ref,
         init_graph();
     } else {
         build_program(is_internal);
+        if (_is_body_program) {
+            // To skip empty if (condition) subgraph
+            bool can_be_optimized = true;
+            for (auto& node : processing_order) {
+                if (node->is_type<input_layout>()) {
+                    continue;
+                } else if (node->is_type<data>()) {
+                    continue;
+                } else if (node->is_output() && !node->has_fused_primitives() &&
+                      node->get_input_layout(0).data_type == node->get_output_layouts(false)[0].data_type) {
```

Comment:
I think it is worth to check that data format is not changed too

Reply:
Applied, thanks